service/s3/s3crypto/aes_gcm.go

```diff
@@ -0,0 +1,162 @@
+package s3crypto
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"io"
+	"io/ioutil"
+)
+
+const (
+	gcmKeySize   = 32
+	gcmNonceSize = 12
+)
+
+// AESGCM Symmetric encryption
+type AESGCM struct {
+	aead  cipher.AEAD
+	nonce []byte
+}
+
+// NewAESGCMRandom create a new AES GCM cipher, but also randomly
+// generates a key and iv.
+//
+// Example:
+//
+// cmkID := "arn to key"
+// kp, _ := s3crypto.NewKMSKeyProvider(session.New(), cmkID, s3crypto.NewJSONMatDesc())
+// cipher, _ := s3crypto.NewAESGCMRandom(kp)
```

Comment:
I think examples need to be indented or they won't show up in the godocs as examples